tutorials/scripting/gdextension/gdextension_cpp_example.rst

```diff
@@ -0,0 +1,652 @@
+we'll save it as ``main.tscn``. We'll come back to that later.
+
+Back in the top-level GDExtension module folder, we're also going to create a
+subfolder called ``src`` in which we'll place our source files.
+
+You should now have ``demo``, ``godot-cpp``, and ``src``
+directories in your GDExtension module.
+
+Your folder structure should now look like this:
+
+.. code-block:: none
+
+    gdextension_cpp_example/
+    |
+    +--demo/                  # game example/demo to test the extension
+    |
+    +--godot-cpp/             # C++ bindings
+    |
+    +--src/                   # source code of the extension we are building
+
+In the ``src`` folder, we'll start with creating our header file for the
+GDExtension node we'll be creating. We will name it ``gdexample.h``:
+
+.. code-block:: C++
+
+    #ifndef GDEXAMPLE_H
+    #define GDEXAMPLE_H
+
+    #include <godot_cpp/classes/sprite2d.hpp>
+
+    namespace godot {
+
+    class GDExample : public Sprite2D {
+        GDCLASS(GDExample, Sprite2D)
+
+    private:
+        float time_passed;
+
+    protected:
+        static void _bind_methods();
+
+    public:
+        GDExample();
+        ~GDExample();
+
+        void _init(); // our initializer called by Godot
+
+        void _process(float delta);
+    };
+
+    }
+
+    #endif
+
+There are a few things of note to the above. We include ``sprite2d.hpp`` which
+contains bindings to the Sprite2D class. We'll be extending this class in our
+module.
+
+We're using the namespace ``godot``, since everything in GDExtension is defined
+within this namespace.
+
+Then we have our class definition, which inherits from our Sprite2D through a
+container class. We'll see a few side effects of this later on. The
+``GDCLASS`` macro sets up a few internal things for us.
+
+After that, we declare a single member variable called ``time_passed``.
+
+In the next block we're defining our methods, we have our constructor
+and destructor defined, but there are two other functions that will likely look
+familiar to some, and one new method.
+
+The first is ``_bind_methods``, which is a static function that Godot will
+call to find out which methods can be called and which properties it exposes. 
+The second is our ``_process`` function, which will work exactly the same 
+as the ``_process`` function you're used to in GDScript. The third is 
+our ``_init`` function which is called after Godot has properly set up
+our object.
+
+Let's implement our functions by creating our ``gdexample.cpp`` file:
+
+.. code-block:: C++
+
+    #include "gdexample.h"
+    #include <godot_cpp/core/class_db.hpp>
+
+    using namespace godot;
+
+    void GDExample::_bind_methods() {
+        ClassDB::bind_method(D_METHOD("_process"), &GDExample::_process);
+    }
+
+    GDExample::GDExample() {
+        // initialize any variables here
+        time_passed = 0.0;
+    }
+
+    GDExample::~GDExample() {
+        // add your cleanup here
+    }
+
+    void GDExample::_process(float delta) {
```

Comment:
I think process works differently now. Process is now defined as a virtual in `node.hpp`: `virtual void _process(double delta);`.

So you now simply override it. You also don't need to bind it, it's already bound.

Reply:
It's also important to call `_process` on the super class or any logic in the super classes will be skipped.

The same applies to a bunch of other functions, just have a look at `node.hpp`.